File: src/main/java/com/dentistPlus/dentist_backend/model/Appointment.java
package com.dentistPlus.dentist_backend.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.Data;

import java.time.LocalDate;
import java.time.LocalTime;

@Data
@Entity
@Table(name = "appointment")
public class Appointment {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;
    @ManyToOne
    @JoinColumn(name = "patient_id")
    private Patient patient;
    @ManyToOne
    @JoinColumn(name = "procedure_id")
    private Procedure procedure;
    private LocalTime time;
    private LocalDate date;

    public Appointment() {
    }

    public void setValues(Appointment appointment) {
        this.setPatient(appointment.getPatient());
        this.setProcedure(appointment.getProcedure());
        this.setTime(appointment.getTime());
        this.setDate(appointment.getDate());
    }
}

File: src/model/Appointment.ts
import mongoose, { Schema } from 'mongoose';

export interface Appointment {
  date: string;
  time: string;
  patientId: string;
  procedureId: string;
}

export interface AppointmentModel extends Appointment, Document {};

const appointmentSchema = new mongoose.Schema({
  date: { type: String, required: true },
  time: { type: String, required: true },
  patientId: { type: Schema.Types.ObjectId, required: true, ref: 'Patient' },
  procedures: { type: Schema.Types.ObjectId, required: true, ref: 'Procedure' }
})

export default mongoose.model<AppointmentModel>('Appointment', appointmentSchema);